File: model/src/main/java/com/tp/holdem/model/game/Player.java
package com.tp.holdem.model.game;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Player {
	private int number;

	private int minimumBet;
	private int maximumBet;
	private int betAmount;
	private int chipsAmount;

	private boolean inGame;
	private boolean allIn;
	private boolean folded;
	private boolean dealerButton;
	private boolean bigBlind;
	private boolean smallBlind;

	private String name;

	@Builder.Default
	private List<Moves> possibleMoves = new ArrayList<>();
	@Builder.Default
	private List<Card> hand = new ArrayList<>();
	private HandRank handRank;

	public void addCard(final Card card) {
		hand.add(card);
	}

	public boolean playing() {
		return isInGame() && !isFolded();
	}
}

File: model/src/main/kotlin/com/tp/holdem/model/Player.kt
package com.tp.holdem.model

import com.tp.holdem.common.model.Moves
import io.vavr.collection.List

data class Player(
        val number: Int = 0,

        val minimumBet: Int = 0,
        val maximumBet: Int = 0,
        val betAmount: Int = 0,
        val betAmountThisPhase: Int = 0,
        val chipsAmount: Int = 0,

        val inGame: Boolean = false,
        val allIn: Boolean = false,
        val folded: Boolean = false,

        val name: String = "Player$number",
        val possibleMoves: List<Moves> = List.empty(),
        val hand: List<Card> = List.empty()
) {
    companion object {
        fun numbered(number: Int): Player {
            return Player(
                    name = "Player$number",
                    number = number
            )
        }
    }
}
